internal/provider/resource_application_deployment.go

```diff
@@ -90,15 +120,27 @@ func (r *applicationDeploymentResource) Create(ctx context.Context, req resource
 	applicationURL := fmt.Sprintf("%s/applications/%v", r.provider.client.ApiURL, data.Application.ValueString())
 	environmentURL := fmt.Sprintf("%s/environments/%v", r.provider.client.ApiURL, data.Environment.ValueString())
 
+	// we count if there is at least one authentication defined for these application and environment
+	authenticationCount := 0
 	// We check if Application Principal exists for this environment and application
-	ApplicationPrincipalFindByApplicationAndEnvironmentResponse, err := r.provider.client.FindApplicationPrincipalByApplicationAndEnvironment(applicationURL, environmentURL)
+	applicationPrincipalsResponse, err := r.provider.client.FindApplicationPrincipalByApplicationAndEnvironment(applicationURL, environmentURL)
 	if err != nil {
 		resp.Diagnostics.AddError("Error querying for Application Principal for this application and environment", fmt.Sprintf("Error message: %s", err.Error()))
 		return
 	}
-	// We do not allow creating Application Deployment if there is no Application Principal because we can't start the connector without it
-	if len(ApplicationPrincipalFindByApplicationAndEnvironmentResponse.Embedded.ApplicationPrincipalResponses) == 0 {
-		resp.Diagnostics.AddError("Error from Terraform Provider validation", "Please first create Application Principal for this application and environment")
+	authenticationCount += len(applicationPrincipalsResponse.Embedded.ApplicationPrincipalResponses)
+	if isKSML(data.Type.ValueString()) {
```

Comment:
```suggestion
	if !data.Type.IsNull() && !data.Type.IsUnknown() && isKSML(data.Type.ValueString()) {
```

The `type` field is defined as `Optional: true` in the schema (line 73), which means `data.Type` can be null. Calling `ValueString()` on a null value will panic. The code must check if the value is null/unknown before calling `ValueString()`.